packages/block-editor/src/components/block-popover/inbetween.js

```diff
@@ -91,66 +91,53 @@ function BlockPopoverInbetween( {
 		};
 	}, [ previousElement, nextElement, isVertical ] );
 
-	const getAnchorRect = useCallback( () => {
+	const popoverAnchor = useMemo( () => {
 		if ( ( ! previousElement && ! nextElement ) || ! isVisible ) {
-			return {};
+			return undefined;
 		}
 
 		const { ownerDocument } = previousElement || nextElement;
 
-		const previousRect = previousElement
-			? previousElement.getBoundingClientRect()
-			: null;
-		const nextRect = nextElement
-			? nextElement.getBoundingClientRect()
-			: null;
-
-		if ( isVertical ) {
-			if ( isRTL() ) {
-				return {
-					top: previousRect ? previousRect.bottom : nextRect.top,
-					left: previousRect ? previousRect.right : nextRect.right,
-					right: previousRect ? previousRect.left : nextRect.left,
-					bottom: nextRect ? nextRect.top : previousRect.bottom,
-					height: 0,
-					width: 0,
-					ownerDocument,
-				};
-			}
-
-			return {
-				top: previousRect ? previousRect.bottom : nextRect.top,
-				left: previousRect ? previousRect.left : nextRect.left,
-				right: previousRect ? previousRect.right : nextRect.right,
-				bottom: nextRect ? nextRect.top : previousRect.bottom,
-				height: 0,
-				width: 0,
-				ownerDocument,
-			};
-		}
-
-		if ( isRTL() ) {
-			return {
-				top: previousRect ? previousRect.top : nextRect.top,
-				left: previousRect ? previousRect.left : nextRect.right,
-				right: nextRect ? nextRect.right : previousRect.left,
-				bottom: previousRect ? previousRect.bottom : nextRect.bottom,
-				height: 0,
-				width: 0,
-				ownerDocument,
-			};
-		}
-
 		return {
-			top: previousRect ? previousRect.top : nextRect.top,
-			left: previousRect ? previousRect.right : nextRect.left,
-			right: nextRect ? nextRect.left : previousRect.right,
-			bottom: previousRect ? previousRect.bottom : nextRect.bottom,
-			height: 0,
-			width: 0,
 			ownerDocument,
+			getBoundingClientRect() {
+				const prevRect = previousElement
+					? previousElement.getBoundingClientRect()
+					: null;
+				const nextRect = nextElement
+					? nextElement.getBoundingClientRect()
+					: null;
+
+				let left = 0;
+				let top = 0;
+
+				if ( isVertical ) {
+					// vertical
+					top = prevRect ? prevRect.bottom : nextRect.top;
+
+					if ( isRTL() ) {
+						// vertical, rtl
+						left = prevRect ? prevRect.right : nextRect.right;
+					} else {
+						// vertical, ltr
+						left = prevRect ? prevRect.left : nextRect.left;
+					}
+				} else {
+					top = prevRect ? prevRect.top : nextRect.top;
+
+					if ( isRTL() ) {
+						// non vertical, rtl
+						left = prevRect ? prevRect.left : nextRect.right;
+					} else {
+						// non vertical, ltr
+						left = prevRect ? prevRect.right : nextRect.left;
+					}
+				}
+
+				return new window.DOMRect( left, top, 0, 0 );
```

Comment:
Change no. 2: moved the remaining logic for creating a rect to the `getBoundingClientRect()` property.

The main two changes here are:

1. Return a proper `DOMRect` object, instead of a manually created one
2. Simplify the logic by removing unnecessary calculations

The removal of unnecessary calculations is based on the fact that currently the resulting rect would always have a `0` `width` and `height` anyway! This is backed by a few reasons:

 - the current implementation on `trunk` always sets `width` and `height` to `0`, which [take the priority in the `Popover` component when computing the final anchor rect](https://github.com/WordPress/gutenberg/blob/4e7508b74e08f03f61b56ecafa6d57fdfedfb3e7/packages/components/src/popover/utils.js#L206-L211)
 - even before the refactor in #40740, the `Popover` [component used to disregard the `right` and `bottom` props when calculating its anchor](https://github.com/WordPress/gutenberg/blob/1e2f974385043cd2ce41c42c1f9638f7b24a43a5/packages/components/src/popover/utils.js#L390-L395)
Even before the refactor